src/libguac/protocol.c

```diff
@@ -74,6 +75,138 @@ ssize_t __guac_socket_write_length_double(guac_socket* socket, double d) {
 
 }
 
+/**
+ * Converts the image data on a Cairo surface to a JPEG image and sends it
+ * as a base64-encoded transmission on the socket.
+ * 
+ * @param socket
+ *     The guac_socket connection to use.
+ * 
+ * @param surface
+ *     A cairo surface containing the image data to send. 
+ * 
+ * @return
+ *     Zero on success, non-zero on error.
+ */
+static int __guac_socket_write_length_jpeg(guac_socket* socket, cairo_surface_t* surface) {
+
+    /* Get image surface properties and data */
+    cairo_format_t format = cairo_image_surface_get_format(surface);
+
+    if (format != CAIRO_FORMAT_RGB24) {
+        guac_error = GUAC_STATUS_INTERNAL_ERROR;
+        guac_error_message = "Invalid Cairo image format. Unable to create JPEG.";
+        return -1;
+    }
+
+    int quality = 80; /* JPEG compression quality setting */
+    int width = cairo_image_surface_get_width(surface);
+    int height = cairo_image_surface_get_height(surface);
+    int stride = cairo_image_surface_get_stride(surface);
+    unsigned char* data = cairo_image_surface_get_data(surface);
+
+    /* Flush pending operations to surface */
+    cairo_surface_flush(surface);
+
+    /* Prepare JPEG bits */
+    struct jpeg_compress_struct cinfo;
+    struct jpeg_error_mgr jerr;
+    cinfo.err = jpeg_std_error(&jerr);
+    jpeg_create_compress(&cinfo);
+
+    unsigned char *mem = NULL;
+    unsigned long mem_size = 0;
+    jpeg_mem_dest(&cinfo, &mem, &mem_size);
+
+    cinfo.image_width = width; /* image width and height, in pixels */
+    cinfo.image_height = height;
+    cinfo.arith_code = TRUE;
+
+#ifdef JCS_EXTENSIONS
```

Comment:
This is something defined within libjpeg's headers if turbo is supported then? If so, I'm surprised (but happy!) that there is no need to add tests to `configure.ac` for this.